legend-sdlc-generation-service/src/test/java/org/finos/legend/sdlc/generation/service/TestServiceRunnerExecution.java

```diff
@@ -0,0 +1,289 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.sdlc.generation.service;
+
+import io.github.classgraph.ClassGraph;
+import org.eclipse.collections.api.factory.Lists;
+import org.finos.legend.engine.language.pure.compiler.Compiler;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.dsl.service.execution.ServiceRunner;
+import org.finos.legend.engine.language.pure.dsl.service.execution.ServiceRunnerInput;
+import org.finos.legend.engine.language.pure.grammar.from.PureGrammarParser;
+import org.finos.legend.engine.plan.execution.nodes.helpers.platform.JavaHelper;
+import org.finos.legend.engine.plan.execution.result.serialization.SerializationFormat;
+import org.finos.legend.engine.plan.generation.transformers.LegendPlanTransformers;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElement;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.service.Service;
+import org.finos.legend.pure.runtime.java.compiled.compiler.MemoryFileManager;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import javax.tools.DiagnosticCollector;
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.ToolProvider;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class TestServiceRunnerExecution
```

Comment:
Why is there a separate class for this? Why isn't this added to TestServiceExecutionGenerator?

Reply:
TestServiceExecutionGenerator tests 'execute' methods which are specific to each service. I added this class to test execution of different services using the common ServiceRunner API and also do some asserts on code generated exception messages (in run method). One other difference is that I wanted to improve readability of the test by using grammar to define the model instead of JSON.

Reply:
TestServiceExecutionGenerator is intended to test that ServiceExecutionGenerator behaves correctly. That includes the generated 'execute' methods which are specific to each service, but is not limited to that. If there are new API methods being added, they should also be tested in TestServiceExecutionGenerator. And if you want to improve readability and add more service definitions to be tested, by all means do so. But it's better to improve the existing test, rather than adding a new test.

Reply:
Done